services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/authentication/PersonalAccessTokenManagerImpl.java

```diff
@@ -125,8 +125,12 @@ public AccessTokenGenerationResponse issueToken(UserInfo userInfo, String access
 
 		try {
 			record = personalAccessTokenDao.createTokenRecord(record);
-		} catch (DuplicateKeyException e) {
-			throw new IllegalArgumentException(DUPLICATE_TOKEN_NAME_MSG, e);
+		} catch (IllegalArgumentException e) {
+			if (e.getMessage().contains("org.springframework.dao.DuplicateKeyException")) {
```

Comment:
In the interest of *strong typing* I would suggest an alternative fix, something like:
Define a class org.sagebionetworks.DuplicateKeyException that extends IllegalArgumentException.
In DBOBasicDaoImpl.insert, catch org.springframework.dao.DuplicateKeyException and throw org.sagebionetworks.DuplicateKeyException.
Here, in PersonalAccessTokenManagerImpl, catch org.sagebionetworks.DuplicateKeyException and throw an exception (either  org.sagebionetworks.DuplicateKeyException or IllegalArgumentException) with DUPLICATE_TOKEN_NAME_MSG (as currently shown at line 130).

Reply:
I found elsewhere we're using `e.getCause() instanceof org.springframework.dao.DuplicateKeyException` to conditionally handle this type of error. I've switched my implementation to use this. Creating a new exception and changing the behavior of DBOBasicDaoImpl would have likely been a much larger change.